scripts/install_dict.py

```diff
@@ -0,0 +1,190 @@
+    parser = argparse.ArgumentParser(prog='install_dict',
+                                     description=desc)
+    parser.add_argument('-l', '--list', action='store_true',
+                        help="Display the list of available languages.")
+    parser.add_argument('languages', nargs='*',
+                        help="A list of languages to install.")
+    return parser
+
+
+def print_list(languages):
+    for lang in languages:
+        print(lang.code, lang.name, sep='\t')
+
+
+def valid_languages():
+    """Return a mapping from valid language codes to their names."""
+    option = configdata.DATA['spellcheck.languages']
+    return option.typ.valtype.valid_values.descriptions
+
+
+def language_list_from_api():
+    """Return a JSON with a list of available languages from Google API."""
+    listurl = urllib.parse.urljoin(API_URL, '?format=JSON')
+    response = urllib.request.urlopen(listurl)
+    # A special 5-byte prefix must be stripped from the response content
+    # See: https://github.com/google/gitiles/issues/22
+    #      https://github.com/google/gitiles/issues/82
+    json_content = response.read()[5:]
+    entries = json.loads(json_content.decode('utf-8'))['entries']
+    return entries
+
+
+def available_languages():
+    """Return a list of Language objects of all available languages."""
+    lang_map = valid_languages()
+    api_list = language_list_from_api()
+    dict_re = re.compile(r"""
+        (?P<filename>(?P<dict>[a-z]{2}(-[A-Z]{2})?).*)\.bdic
+    """, re.VERBOSE)
+    code2file = {}
+    for lang in api_list:
+        match = dict_re.match(lang['name'])
+        if match is not None:
+            code2file[match.group('dict')] = match.group('filename')
+    return [
+        Language(code, name, code2file[code])
+        for code, name in lang_map.items()
+    ]
+
+
+def download_dictionary(url, dest):
+    """Download a decoded dictionary file."""
+    response = urllib.request.urlopen(url)
+    decoded = base64.decodebytes(response.read())
+    with open(dest, 'bw') as dict_file:
+        dict_file.write(decoded)
+
+
+def filter_languages(languages, selected):
+    """Filter a list of languages based on an inclusion list.
+
+    Args:
+        languages: a list of languages to filter
+        selected: a list of keys to select
+    """
+    filtered_languages = []
+    for language in languages:
+        if language.code in selected:
+            filtered_languages.append(language)
+            selected.remove(language.code)
+    if selected:
+        raise InvalidLanguageError(selected)
+    return filtered_languages
+
+
+def install_lang(lang):
+    """Install a single lang given by the argument."""
+    print('Installing {}: {}'.format(lang.code, lang.name))
+    lang_url = urllib.parse.urljoin(API_URL, lang.file_path, '?format=TEXT')
+    if not os.path.isdir(spell.dictionary_dir()):
+        warn_msg = 'WARN: {} does not exist, creating the directory'
```

Comment:
No need for the `WARN: ` prefix here I think - it's not like something bad happened :wink: